common/common.go

```diff
@@ -141,10 +143,30 @@ func newHTTPClient() *http.Client {
 	return &http.Client{}
 }
 
+// Stubbed methods below
+// this is done so that this function can be stubbed
+// for unit testing
+
 // Exists Checks if a file exists on the filesystem
 var Exists = func(name string) bool {
 	if _, err := os.Stat(name); os.IsNotExist(err) {
 		return false
 	}
 	return true
 }
+
+// OsHostname is a wrapper to get os hostname
+var OsHostname = func() (name string, err error) {
+	return os.Hostname()
+}
+
+// ReadFile is a wrapper to read file
+var ReadFile = func(file string) ([]byte, error) {
```

Comment:
I'd prefer to not stub out things we don't need to (readfile, exists maybe), also these shouldn't be common if no other code uses them, just keep it private next to the code that uses it.

Reply:
i introduced stubs for these small utility functions because of a series of reasons.
- to test presence of package manager binaries, i could have either created fake binaries on the native file system or a fake binary on a test file system or i could stub the exists method itself. i chose to stub because creating fake binaries would not work system where the actual binary is already present(though there are ways to get around it), for simplicity and less maintenance i stubbed the "exists" method. with this approach i had this concern that we are not testing existence of package manager binary; but i ignored it since we can safely assume that apt would be present on an apt based system and if it not then the agent must crash anyways. it is on the customer to fix it(correct me if i am wrong here).

- for readfile; i do feel i could have just used the native file system to create the files, but since i was already using the fake filesystem for package binaries; i did not want to mix things up.




> these shouldn't be common if no other code uses them, just keep it private next to the code that uses it.

most of it like anything related to fetcher is already moved to a different package. i would still like to keep these utility functions public since these are accessed by different package. will change this to "utility"?